chainio/txmgr/txmgr.go

```diff
@@ -48,29 +42,30 @@ type EthBackend interface {
 }
 
 type SimpleTxManager struct {
-	backend   EthBackend
-	signerFn  signerv2.SignerFn
-	log       logging.Logger
-	sender    common.Address
-	contracts map[common.Address]*bind.BoundContract
+	txSender txsender.TxSender
+	backend  EthBackend
+	signerFn signerv2.SignerFn
+	log      logging.Logger
+	sender   common.Address
 }
 
 var _ TxManager = (*SimpleTxManager)(nil)
 
 // NewSimpleTxManager creates a new simpleTxManager which can be used
 // to send a transaction to smart contracts on the Ethereum node
 func NewSimpleTxManager(
+	txSender txsender.TxSender,
 	backend EthBackend,
```

Comment:
Think we should get rid of this ethbackend interface now, and just use eth.Client interface everywhere (pretty sure eth.Client contains all of these EthBackend methods).

Can you just add a TODO here above `backend EthBackend` to investigate whether we can just replace it by the eth.Client. Let's do it in another PR though to not make this PR too big.